receiver/sqlserverreceiver/config.go

```diff
@@ -24,4 +26,17 @@ import (
 type Config struct {
 	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
 	metadata.MetricsBuilderConfig           `mapstructure:",squash"`
+	InstanceName                            string `mapstructure:"instance_name"`
+	ComputerName                            string `mapstructure:"computer_name"`
+}
+
+func (cfg *Config) Validate() error {
+	if cfg.InstanceName != "" && cfg.ComputerName == "" {
+		return fmt.Errorf("missing 'instance name', both instance name and computer name must be set")
+	}
+	if cfg.InstanceName == "" && cfg.ComputerName != "" {
+		return fmt.Errorf("missing 'computer_name', both instance name and computer name must be set")
+	}
```

Comment:
Based on the description in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20247, `identified by the computer name or the computer name followed by an instance name `, I was expecting the instance name to optional regardless. Are you sure this is right?

Reply:
Either way, the error message makes it sound like both fields are always required, but you are only enforcing that both are specified if at least one is.

Reply:
I was testing a bunch of PDH string formats. To collect metrics from a named instance, the computer name and instance name must specified. I tried without the computer name and it did not work. These 2 if statements enforces that if one is used, then so is the other when collecting from a named instance. If both are empty, or filled then validate will return nil. So they are both always required when collecting from a named instance.

I do believe if you are not collecting from a named instance, that you can still have a computer name attached. In that case, this would then not work if you wanted the computer.name resource attribute.

Reply:
👍 Consider updating the error messages to something like: `'computer_name' may not be specified without 'instance name'`